tests/test_script_queue.py

```diff
@@ -348,6 +349,126 @@ def make_add_data(location, locationSalIndex=0):
+                descr="test_add_log_level",
+                timeout=START_TIMEOUT)
+            await self.assert_next_queue(running=False, salIndices=[I0])
+
+            script_state = await script_remote.evt_state.next(flush=False, timeout=START_TIMEOUT)
+            self.assertEqual(script_state.state, ScriptState.UNCONFIGURED)
+            script_state = await script_remote.evt_state.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertEqual(script_state.state, ScriptState.CONFIGURED)
+
+            # check initial log level
+            data = await script_remote.evt_logLevel.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertEqual(data.level, logging.INFO)
+
+            # if log_level != 0 check final log level
+            # else check that no second log level was output
+            if log_level != 0:
+                data = await script_remote.evt_logLevel.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(data.level, log_level)
+            else:
+                with self.assertRaises(asyncio.TimeoutError):
+                    await script_remote.evt_logLevel.next(flush=False, timeout=0.01)
+
+            # make sure the script queue knows that the script is ready to run
+            await self.wait_runnable(I0)
+
+            # enable the queue and let it run
+            await self.remote.cmd_resume.start(timeout=STD_TIMEOUT)
+            await self.assert_next_queue(enabled=True, running=True, currentSalIndex=I0)
+            await self.assert_next_queue(enabled=True, running=True, pastSalIndices=[I0])
+
+    async def test_add_nonzero_log_level(self):
+        """Test addding a script with a non-zero log level."""
+        # pick a level that does not match the default
+        # to make it easier to see that the level has changed
+        log_level = logging.INFO - 1
+        await self.check_add_log_level(log_level=log_level)
+
+    async def test_add_zero_log_level(self):
+        """Test addding a script with log level 0, meaning don't change it."""
+        await self.check_add_log_level(log_level=0)
+
+    async def test_add_and_pause(self):
+        """Test adding a script with a pause checkpoint.
+        """
+        await self.assert_next_queue(enabled=False, running=True)
+        await self.remote.cmd_enable.start(timeout=STD_TIMEOUT)
+        await self.assert_next_queue(enabled=True, running=True)
+
+        async with salobj.Remote(domain=self.queue.domain, name="Script", index=I0) as script_remote:
+            await self.remote.cmd_add.set_start(
+                pauseCheckpoint="start",
+                isStandard=False,
+                path="script1",
+                config="",
+                location=Location.LAST,
+                descr="test_add",
+                timeout=START_TIMEOUT)
+            await self.assert_next_queue(running=True, salIndices=[I0])
+            await self.assert_next_queue(running=True, currentSalIndex=I0)
```

Comment:
It looks like there's a bunch of "startup" boilerplate in these tests. Is there any way to refactor it out, possibly by passing a `**kwargs` argument to a `check_startup` method?

Reply:
I agree and I've looked at that but I haven't thought of anything I like well enough to switch.

The problem is that the tests are subtly different from each other (some start a queue and leave it running, some start a queue and pause it). I guess that could be a flag. I'll have another look.

The most verbose bits are making the data for the add command, and I've definitely thought of a test method `make_add_data` (instead of test-specific functions that I now use). But there's something to be said for being explicit.

Reply:
I'd like to leave things as they are. A bit verbose, perhaps, but every step is realistic and I just have not thought of a refactor that is worth the work.

Reply:
Ok. This is like my experience with the ts_FiberSpectrograph tests: the tests felt much more verbose than they needed to be, but I couldn't come up with a way to clean them up. It might be worth us discussing some options, since we're going to need more CSC tests in the future.